ui/page/staking/auto_ticket_modal.go

```diff
@@ -0,0 +1,234 @@
+package staking
+
+import (
+	"strconv"
+
+	"gioui.org/layout"
+	"gioui.org/text"
+	"gioui.org/widget"
+
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/page/components"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+const ticketBuyerModalID = "staking_modal"
+
+type ticketBuyerModal struct {
+	*load.Load
+
+	vspIsFetched bool
+
+	settingsSaved func()
+	cancelFunc    func()
+
+	modal           decredmaterial.Modal
+	cancel          decredmaterial.Button
+	saveSettingsBtn decredmaterial.Button
+
+	balToMaintainEditor decredmaterial.Editor
+
+	accountSelector *components.AccountSelector
+	vspSelector     *vspSelector
+}
+
+func newTicketBuyerModal(l *load.Load) *ticketBuyerModal {
+	tb := &ticketBuyerModal{
+		Load: l,
+
+		cancel:          l.Theme.OutlineButton("Cancel"),
+		saveSettingsBtn: l.Theme.Button("Save"),
+		modal:           *l.Theme.ModalFloatTitle(),
+	}
+
+	tb.balToMaintainEditor = l.Theme.Editor(new(widget.Editor), "Balance to maintain (DCR)")
+	tb.balToMaintainEditor.Editor.SetText("")
+	tb.balToMaintainEditor.Editor.SingleLine = true
+
+	tb.saveSettingsBtn.SetEnabled(false)
+
+	tb.vspIsFetched = len(l.WL.MultiWallet.VspList) > 0
+
+	return tb
+}
+
+func (tb *ticketBuyerModal) SettingsSaved(settingsSaved func()) *ticketBuyerModal {
+	tb.settingsSaved = settingsSaved
+	return tb
+}
+
+func (tb *ticketBuyerModal) CancelSave(cancel func()) *ticketBuyerModal {
```

Comment:
```suggestion
func (tb *ticketBuyerModal) OnCancel(cancel func()) *ticketBuyerModal {
```

`tb.cancelFunc` should also probably be `tb.onCancel`.